vscode_extension/README.md

```diff
@@ -1,36 +1,114 @@
+
+### Architecture-as-Code Language Server (LSP)
+
+The AaC tool implements the Language Server Protocol (LSP), and the AaC VSCode Extension integrates with the tool's LSP features to bring common programming IDE features to the AaC DSL and AaC files.
+
+#### Rename
+
+AaC supports renaming, which performs a workspace-wide rename of a symbol.
+
+AaC currently only supports renaming Definition names -- an important feature given that Definition names are used to reference other modeled components and behaviors.
+
+![LSP Rename Example](./images/readme/aac_lsp_rename.gif)
+
+#### Go To/Find All References
+
+AaC supports finding references, which performs a workspace-wide search for any symbols referencing the targetted item.
+
+![LSP Go to/Find All References Example](./images/readme/aac_lsp_find_all_references.gif)
+
+#### Go To Definition
+
+AaC supports go to definitions, which performs a workspace-wide search for the declaration of a particular symbol.
+
+![LSP Go to Definition Example](./images/readme/aac_lsp_goto_definition.gif)
+
+#### Diagnostics
+
+AaC provides real-time validation and diagnostics. These diagnostics are driven by the AaC tool's core command `validate`.
+
+![LSP Diagnostics Example](./images/readme/aac_lsp_diagnostics.gif)
+
+#### Document Linking
+
+Document linking allows the IDE to provide clickable navigation links between AaC files.
+
+![LSP Document Linking Example](./images/readme/aac_lsp_document_link.gif)
+
+#### Hover (Preview)
+
+AaC provides a preview of AaC definitions when you hover any reference to them.
 
-### Architecture-as-Code Language Server Protocol (LSP)
-The AaC VSCode extension integrates with the AaC python package to provide IDE language features for AaC files.
+![LSP Hover Example](./images/readme/aac_lsp_hover.gif)
+
+#### Code Completion
+
+AaC supports code completion in order to ease developers' experience when modeling in AaC.
+
+Currently, the support for code completion is limited, but further expansion is planned for the future.
+
+![LSP Code Completion Example](./images/readme/aac_lsp_code_completion.gif)
+
+### Visual Definition Editor (Alpha)
 
-### Visual Definition Editor
 The AaC VSCode extension can leverage the AaC python package's `rest-api` plugin to provide VSCode users with an in-IDE visual editor for AaC definitions.
 
 Before getting started, make sure that you have the AaC `rest-api` plugin running in your project directory.
 ```bash
 aac rest-api --host <hostname|localhost> --port <8000>
 ```
 
+![Visual Editor Example](./images/readme/aac_visual_editor.gif)
+
+## Getting Started
+
+The AaC VSCode extension does not require configuration as long as the `aac`
+Python tool is in your system `PATH`. Nonetheless, there are configuration options
+available, so for detailed instructions on configuring the AaC VSCode Extension,
+refer to [this](https://jondavid-black.github.io/AaC/docs/vscode_extension) page.
```

Comment:
I would just name the title of the page rather than say this.

Reply:
done.